lib/Backend/arm64/LegalizeMD.cpp

```diff
@@ -858,17 +858,17 @@ bool LegalizeMD::LegalizeDirectBranch(IR::BranchInstr *branchInstr, uintptr_t br
     uintptr_t labelOffset = branchInstr->GetTarget()->GetOffset();
     Assert(labelOffset); //Label offset must be set.
 
-    intptr_t offset = labelOffset - branchOffset;
+    intptr_t wordOffset = intptr_t(labelOffset - branchOffset) / 4;
     //We should never run out of 26 bits which corresponds to +-64MB of code size.
-    AssertMsg(IS_CONST_INT26(offset >> 1), "Cannot encode more that 64 MB offset");
+    AssertMsg(IS_CONST_INT26(wordOffset), "Cannot encode more that 64 MB offset");
 
     Assert(!LowererMD::IsUnconditionalBranch(branchInstr));
 
     int limit = 0;
     if (branchInstr->m_opcode == Js::OpCode::TBZ || branchInstr->m_opcode == Js::OpCode::TBNZ)
     {
         // TBZ and TBNZ are limited to 14 bit offsets.
-        if (IS_CONST_INT14(offset))
+        if (IS_CONST_INT14(wordOffset))
         {
```

Comment:
I think you need to fix "limit" here and below as well.

Reply:
I think the limits should be correct as I modified TryInsertBranchIsland to count by 1 instead of MachMaxInstrSize